internal/controller/instaslice_controller.go

```diff
@@ -548,42 +613,72 @@ func (l *RightToLeftPolicy) SetAllocationDetails(profileName string, newStart, s
 	return &inferencev1alpha1.AllocationDetails{}
 }
 
-func (r *InstasliceReconciler) processInstasliceAllocation(ctx context.Context, instasliceName string, podUUID string, allocation inferencev1alpha1.AllocationDetails, req ctrl.Request) (bool, ctrl.Result, error) {
-	if allocation.PodUUID == podUUID {
-		if allocation.Allocationstatus == inferencev1alpha1.AllocationStatusDeleted {
-			deleteResult, errDeletingAllocation := r.deleteInstasliceAllocation(ctx, instasliceName, allocation)
-			if errDeletingAllocation != nil {
-				return true, deleteResult, errDeletingAllocation
-			}
+func (r *InstasliceReconciler) removeInstasliceAllocation(ctx context.Context, instasliceName string, allocation inferencev1alpha1.AllocationDetails) (ctrl.Result, error) {
+	if allocation.Allocationstatus == inferencev1alpha1.AllocationStatusDeleted {
+		deleteResult, errDeletingAllocation := r.deleteInstasliceAllocation(ctx, instasliceName, allocation)
+		if errDeletingAllocation != nil {
+			return deleteResult, errDeletingAllocation
+		}
+	}
+	return ctrl.Result{}, nil
+}
 
-			removeResult, errRemovingFinalizer := r.removeInstaSliceFinalizer(ctx, req)
-			if errRemovingFinalizer != nil {
-				return true, removeResult, errRemovingFinalizer
-			}
-		} else {
-			allocation.Allocationstatus = inferencev1alpha1.AllocationStatusDeleting
+func (r *InstasliceReconciler) setInstasliceAllocationToDeleting(ctx context.Context, instasliceName string, podUUID string, allocation inferencev1alpha1.AllocationDetails) (ctrl.Result, error) {
 
-			var updateInstasliceObject inferencev1alpha1.Instaslice
-			typeNamespacedName := types.NamespacedName{
-				Name:      instasliceName,
-				Namespace: instaSliceOperatorNamespace, // TODO: modify if needed
-			}
-			errRetrievingInstaSlice := r.Get(ctx, typeNamespacedName, &updateInstasliceObject)
-			if errRetrievingInstaSlice != nil {
-				log.FromContext(ctx).Error(errRetrievingInstaSlice, "error getting latest instaslice object")
-				return true, ctrl.Result{Requeue: true}, errRetrievingInstaSlice
+	allocation.Allocationstatus = inferencev1alpha1.AllocationStatusDeleting
+
+	var updateInstasliceObject inferencev1alpha1.Instaslice
+	typeNamespacedName := types.NamespacedName{
+		Name:      instasliceName,
+		Namespace: instaSliceOperatorNamespace, // TODO: modify if needed
```

Comment:
We should have a github issue to track this TODO and refer that issue here.

Reply:
Yes, one of my tasks before we produce a release, is to deploy InstaSlice in namespace `instaslice-operator-system` and test the whole system. Refer : https://github.com/openshift/instaslice-operator/issues/48

Reply:
Cool, can you link the issue here? e.g https://github.com/kubernetes/kubernetes/blob/aa09157014750871407aca4217f4ecf07bc64b7c/pkg/kubelet/cadvisor/util.go#L74